stremio-deeplinks/src/lib.rs

```diff
@@ -140,16 +144,54 @@ impl From<&MetaItemPreview> for MetaItemDeepLinks {
                         "#/metadetails/{}/{}/{}",
                         utf8_percent_encode(&item.r#type, URI_COMPONENT_ENCODE_SET),
                         utf8_percent_encode(&item.id, URI_COMPONENT_ENCODE_SET),
-                        utf8_percent_encode(&video_id, URI_COMPONENT_ENCODE_SET)
+                        utf8_percent_encode(video_id, URI_COMPONENT_ENCODE_SET)
                     )
                 }),
+            player: if item.id.starts_with(YOUTUBE_PREFIX) {
+                Some(())
+            } else {
+                None
+            }
+            .zip(item.behavior_hints.default_video_id.clone())
+            .and_then(|(_, default_video_id)| {
+                // video id are formed like that: yt_id:YT_CHANNEL_ID:YT_VIDEO_ID
+                default_video_id.split(':').nth(2).map(|video_id| {
+                    format!(
+                        "#/player/{}/{}/{}/{}/{}/{}",
+                        utf8_percent_encode(
+                            &base64::encode(
+                                gz_encode(
+                                    serde_json::to_string(&Stream {
+                                        source: StreamSource::YouTube {
+                                            yt_id: video_id.to_string()
```

Comment:
we used to_owned on every other place, change this for consistency

Reply:
Changed